business/istio_status.go

```diff
@@ -169,14 +208,14 @@ func istioCoreComponents() map[string]bool {
 	return components
 }
 
-func (iss *IstioStatusService) getStatusOf(ds []apps_v1.Deployment) (IstioComponentStatus, error) {
+func (iss *IstioStatusService) getStatusOf(wls []workloadInfo) (IstioComponentStatus, error) {
 	statusComponents := istioCoreComponents()
 	isc := IstioComponentStatus{}
 	cf := map[string]bool{}
 
 	// Map workloads there by app name
-	for _, d := range ds {
-		appLabel := labels.Set(d.Spec.Template.Labels).Get("app")
+	for _, wl := range wls {
+		appLabel := wl.Labels.Get("app")
```

Comment:
Note that these constants can be also changed in Istio, so better to read from the main config to get the IstioAppLabel

Reply:
This was changed in https://github.com/kiali/kiali/pull/3544. Istio doesn't change the `app` label in their deployments (https://github.com/istio/istio/issues/29742)